pkg/cvo/status.go

```diff
@@ -347,18 +352,34 @@ func (optr *Operator) syncStatus(ctx context.Context, original, config *configv1
 // unavailable AND the general payload task making progress towards its goal. An operator is given 40 minutes since
 // its last update to go ready, or an hour has elapsed since the update began, before the condition is ignored.
 func convertErrorToProgressing(history []configv1.UpdateHistory, now time.Time, status *SyncWorkerStatus) (reason string, message string, ok bool) {
-	if len(history) == 0 || status.Failure == nil || status.Reconciling || status.LastProgress.IsZero() {
+	if len(history) == 0 || status.Failure == nil || status.Reconciling {
 		return "", "", false
 	}
-	if now.Sub(status.LastProgress) > 40*time.Minute || now.Sub(history[0].StartedTime.Time) > time.Hour {
+	if now.Sub(history[0].StartedTime.Time) > time.Hour {
 		return "", "", false
 	}
 	uErr, ok := status.Failure.(*payload.UpdateError)
 	if !ok {
 		return "", "", false
 	}
 	if uErr.Reason == "ClusterOperatorNotAvailable" || uErr.Reason == "ClusterOperatorsNotAvailable" {
-		return uErr.Reason, fmt.Sprintf("waiting on %s", uErr.Name), true
+		if !uErr.UpgradeSucceeding {
```

Comment:
nit: it's a bit easier for me to read if we avoid the double-negative of `if !something { } else { /* makes me think too hard */ }` and use:

```go
if uErr.UpgradeSucceeding {
  return uErr.Reason, fmt.Sprintf("waiting on %s", uErr.Name), true
} else {
  // !succeeding logic
}
```